### DISCUSSION TITLE
Is there a way to edit styles on an existing template document?

### DISCUSSION THREAD

**Discussion by Suniron** [dolanmiu/docx#2230] [Q&A]
Hey,

I want to edit the style of an existing template document.

For example, I want to change the style of all Heading 1 to a specific color.

I didn't find the way in the documentation..

Thank's a lot for your help!

**Comment by dolanmiu** [comment]
Not possible right now
only possible to edit text in an existing document

**Reply by Suniron** [reply]
Oh 😕. Thanks for your confirmation.

Have you an idea about the way to do that?

**Reply by dolanmiu** [reply]
Currently `docx` allows you to create styles programatically like so:

https://github.com/dolanmiu/docx/blob/master/demo/2-declaritive-styles.ts

Maybe something like this can work for your use case

Why would you want to modify the styles of a document? What is the use case?

**Reply by Suniron** [reply]
I have existing .docx templates that can be uploaded (and patched, this aspect is OK) by users. Uploaded documents should have the headings style (among others) customized with the company's color.

I can't use demo 2 for this use case since it seems to be not possible to merge an existing docx document into a new Document.

**Reply by dolanmiu** [reply]
Is it possible to write the whole thing in code? Or is that unfeasible

**Reply by Suniron** [reply]
Impossible 😱.

Those templates are maintained by a non-developer and we have almost 15 docx templates and even more in the future 😥
